torch/csrc/Exceptions.h

```diff
@@ -385,9 +384,11 @@ auto wrap_pybind_function_impl_(
     HANDLE_TH_ERRORS
     if (release_gil) {
       py::gil_scoped_release no_gil;
-      return c10::guts::invoke(f, std::forward<Arg<Func, Is>>(args)...);
+      return c10::guts::invoke(
+          std::move(f), std::forward<Arg<Func, Is>>(args)...);
```

Comment:
Why should this be `std::move` instead of std::forward?

Reply:
Since f is captured as a value?

Reply:
I think the move here is wrong and leads to potential use of a moved-from object:

- `wrap_pybind_function` is used to bind Python functions to C++ functions via pybind.
- The created lambda may be called multiple times
- After the first call `f` in the lambda is technically in a moved-from state and must no longer be used
- But every successive call will use it.
- That C++ function (i.e. `f`) might be anything, e.g. a capturing lambda which after this move might have "lost" a captured resource

I reverted this part of the change in https://github.com/pytorch/pytorch/pull/116695